cpu/native/can/candev_linux.c

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright (C) 2016 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     native_cpu
+ * @{
+ *
+ * @file
+ * @brief       Implementation of simulated CAN controller driver using SocketCAN on Linux
+ *
+ * @author      Hermann Lelong <hermann@otakeys.com>
+ * @author      Aurelien Gonce <aurelien.gonce@altran.com>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @}
+ */
+
+#if !defined(__linux__)
+#error "MODULE can_linux is only available on Linux"
+#else
+
+#include <string.h>
+#include <errno.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+
+#include <linux/can/raw.h>
+#include <linux/can/error.h>
+
+#include "native_internal.h"
+#include "can/device.h"
+#include "candev_linux.h"
+#include "thread.h"
+#include "mutex.h"
+#include "async_read.h"
+#include "sched.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static int _init(candev_t *candev);
+static int _send(candev_t *candev, const struct can_frame *frame);
+static void _isr(candev_t *candev);
+static int _set(candev_t *candev, canopt_t opt, void *value, size_t value_len);
+static int _get(candev_t *candev, canopt_t opt, void *value, size_t max_len);
+static int _abort(candev_t *candev, const struct can_frame *frame);
```

Comment:
Yes, `abort`, `power_up` and `power_down` make sense only for real CAN devices. The native candev is a bit simplified.